test/unittest/transport/TCPv4Tests.cpp

```diff
@@ -285,6 +284,286 @@ TEST_F(TCPv4Tests, send_to_wrong_interface)
+    sendTransportUnderTest.init();
+
+    Locator_t inputLocator;
+    inputLocator.kind = LOCATOR_KIND_TCPv4;
+    inputLocator.port = 5100;
+    IPLocator::setIPv4(inputLocator, 127, 0, 0, 1);
+    IPLocator::setLogicalPort(inputLocator, 7410);
+
+    Locator_t outputLocator;
+    outputLocator.kind = LOCATOR_KIND_TCPv4;
+    IPLocator::setIPv4(outputLocator, 127, 0, 0, 1);
+    outputLocator.port = 5100;
+    IPLocator::setLogicalPort(outputLocator, 7410);
+
+    {
+        MockReceiverResource receiver(receiveTransportUnderTest, inputLocator);
+        MockMessageReceiver *msg_recv = dynamic_cast<MockMessageReceiver*>(receiver.CreateMessageReceiver());
+        ASSERT_TRUE(receiveTransportUnderTest.IsInputChannelOpen(inputLocator));
+
+        ASSERT_TRUE(sendTransportUnderTest.OpenOutputChannel(outputLocator));
+        octet message[5] = { 'H','e','l','l','o' };
+        bool bOk = false;
+        std::function<void()> recCallback = [&]()
+        {
+            EXPECT_EQ(memcmp(message, msg_recv->data, 5), 0);
+            bOk = true;
+        };
+
+        msg_recv->setCallback(recCallback);
+
+        bool bFinish(false);
+        auto sendThreadFunction = [&]()
+        {
+            bool sent = sendTransportUnderTest.Send(message, 5, outputLocator, inputLocator);
+            while (!bFinish && !sent)
+            {
+                sent = sendTransportUnderTest.Send(message, 5, outputLocator, inputLocator);
+                std::this_thread::sleep_for(std::chrono::milliseconds(100));
+            }
+            EXPECT_TRUE(sent);
+            //EXPECT_TRUE(transportUnderTest.Send(message, 5, outputLocator, inputLocator));
+        };
+
+        senderThread.reset(new std::thread(sendThreadFunction));
+        std::this_thread::sleep_for(std::chrono::seconds(10));
+        bFinish = true;
+        senderThread->join();
+        ASSERT_TRUE(bOk);
+    }
+    ASSERT_TRUE(sendTransportUnderTest.CloseOutputChannel(outputLocator));
+}
+
+TEST_F(TCPv4Tests, send_and_receive_between_blocked_interfaces_ports)
+{
+    LocatorList_t interfaces;
+    if (IPFinder::getAllIPAddress(&interfaces))
+    {
+        Locator_t locator;
+        for (auto& tmpLocator : interfaces)
+        {
+            if (tmpLocator.kind == LOCATOR_KIND_UDPv4 && IPLocator::toIPv4string(tmpLocator) != "127.0.0.1")
+            {
+                locator = tmpLocator;
+                break;
+            }
+        }
+
+        if (IsAddressDefined(locator))
+        {
+            Log::SetVerbosity(Log::Kind::Info);
+            std::regex filter("RTCP(?!_SEQ)");
+            Log::SetCategoryFilter(filter);
+            TCPv4TransportDescriptor recvDescriptor;
+            recvDescriptor.interfaceWhiteList.emplace_back(IPLocator::toIPv4string(locator));
+            recvDescriptor.maxMessageSize = 5;
+            recvDescriptor.sendBufferSize = 5;
+            recvDescriptor.receiveBufferSize = 5;
+            recvDescriptor.add_listener_port(5100);
```

Comment:
User a random port.

Reply:
Changed to use the variable g_default_port